### PR TITLE
Replaces broccoli-static-compiler with broccoli-funnel

### PR DESCRIPTION
**Pull request by twokul**

### CODE REVIEW

File: lib/config/test-config-tree.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 var replaceFeatures    = require('./../replace-features');
 
 module.exports = function getTestConfigTree(opts) {
   var options = opts || {};
 
-  var testConfig = pickFiles(options.libPath || 'tests', {
-    srcDir: '/',
-    files: ['**/*.*'],
+  var testConfig = new Funnel(options.libPath || 'tests', {
+    include: [new RegExp('html')],
```
Comment (rwjblue) [inline lib/config/test-config-tree.js]: Should anchor to the end of the string: `/html$/`

File: lib/ember-dev-test-helpers.js [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-  srcDir: '/',
-  destDir: '/ember-dev',
-  files: ['**/*.js']
-});
+var Funnel = require('broccoli-funnel');
 
 module.exports = function getEmberDevTestHelpers() {
-  return emberDevTestHelpers;
+  return new Funnel('bower_components/ember-dev/addon', {
+    include: [ new RegExp('js') ],
```
Comment (rwjblue) [inline lib/ember-dev-test-helpers.js]: anchor to extension

File: lib/get-build-template-compiler-tree.js [hunk 3]
```diff
@@ -38,7 +37,7 @@ module.exports = function buildTemplateCompilerTree(packages) {
   var compiled = concatenateES6Modules(trees, {
     includeLoader: true,
     bootstrapModule: 'ember-template-compiler',
-    vendorTrees: mergeTrees(vendorTrees),
+    vendorTrees: mergeTrees(vendorTrees, {overwrite: true}),
```
Comment (rwjblue) [inline lib/get-build-template-compiler-tree.js]: We didn't formerly overwrite, why do we need to now?

File: lib/get-es6-package.js [hunk 4]
```diff
@@ -18,8 +17,10 @@ var disableJSHint = config.disableJSHint;
 var disableJSCS   = config.disableJSCS;
 
 function getES6Package(packages, packageName, opts) {
-  var pkg     = packages[packageName];
-  var options = opts || {};
+  var pkg        = packages[packageName];
+  var options    = opts || {};
+  var jsRegExp   = new RegExp('js');
```
Comment (rwjblue) [inline lib/get-es6-package.js]: anchor to extension

File: lib/get-es6-package.js [hunk 5]
```diff
@@ -41,12 +42,12 @@ function getES6Package(packages, packageName, opts) {
...
   /*
-    The list of files to select. This is passed to `pickFiles` below.
+    The list of files to select. This is passed to `Funnel` below.
   */
-  var files = [ '**/*.js'];
+  var files = [ jsRegExp ];
 
   if (pkg.hasTemplates) {
-    files.push('**/*.hbs');
+    files.push(new RegExp('hbs'));
```
Comment (rwjblue) [inline lib/get-es6-package.js]: anchor to extension

File: lib/htmlbars-package.js [hunk 6]
```diff
@@ -16,20 +16,19 @@ var buildConfig  = require('./config/build-config');
...
 */
 module.exports = function htmlbarsPackage(packageName, _options) {
-  var options = _options || {};
-  var fileGlobs = [];
+  var options  = _options || {};
+  var includes = [];
 
   if (!options.singleFile) {
-    fileGlobs.push(packageName + '/**/*.js');
+    includes.push(new RegExp('js'));
```
Comment (rwjblue) [inline lib/htmlbars-package.js]: anchor to extension

File: lib/minify-source-tree.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ module.exports = function minifySourceTree(tree, options) {
     throw new Error('Tree must be defined');
   }
 
-  var minifiedTree = moveFile(tree, {
+  var minifiedTree = new Funnel(tree, {
     srcFile:  options.srcFile,
```
Comment (rwjblue) [inline lib/minify-source-tree.js]: I do not think this works (`srcFile` isn't an option to Funnel).

File: tests/concatatinate-es6-modules-test.js [hunk 8]
```diff
@@ -18,14 +18,12 @@ var fixturesLoaderPath = path.join(__dirname, './fixtures/loader');
 describe('concatenate-es6-modules', function() {
   var builder;
 
-  var testTree = pickFiles(fixturesTestPath, {
-    srcDir: '/',
-    files: ['**/*.js'],
+  var testTree = new Funnel(fixturesTestPath, {
+    includes: [ new RegExp('js') ],
```
Comment (rwjblue) [inline tests/concatatinate-es6-modules-test.js]: anchor to extension

File: lib/get-build-template-compiler-tree.js (hunk 3 above)
Comment (twokul) [inline lib/get-build-template-compiler-tree.js]: dependencies step on each other if we don't use overwrite, I'm unsure why exactly.

File: tests/replace-version-test.js [hunk 9]
```diff
@@ -17,9 +17,8 @@ var fixturesPath = path.join(__dirname, './fixtures/versioned-files');
 describe('replace-version', function() {
   var builder;
 
-  var trees = pickFiles(fixturesPath, {
-    srcDir: '/',
-    files: ['**/*.js', '**/*.json'],
+  var trees = new Funnel(fixturesPath, {
+    includes: [ new RegExp('js'), new RegExp('json') ],
```
Comment (rwjblue) [inline tests/replace-version-test.js]: anchor to extension

**Pull request merged**